torch/_dynamo/variables/dicts.py

```diff
@@ -344,33 +337,25 @@ def call_method(
             self.items.clear()
             return ConstantVariable.create(None)
         elif name == "update" and self.is_mutable():
-            is_args_supported = len(args) == 1 and isinstance(
-                args[0],
-                (
-                    ConstDictVariable,
-                    ListVariable,
-                    TupleVariable,
-                    ListIteratorVariable,
-                    variables.IteratorVariable,
-                    UserDefinedObjectVariable,
-                ),
-            )
-
-            is_kwargs_supported = len(kwargs) > 0 and len(args) == 0
-
-            if is_args_supported or is_kwargs_supported:
```

Comment:
`BuiltinVariable.call_custom_dict` takes care of all these instance checks and fail with `unimplemented` for unsupported cases, so we don't need to duplicate the check here. Checking here also restricted this from hitting the new `elif hasattr(arg, "value") and isinstance(arg.value, MutableMapping):` branch for more general custom dict support.
 
https://github.com/pytorch/pytorch/blob/604c5a9e29eb9939d3b3538fd5b04753d7aeeef5/torch/_dynamo/variables/builtin.py#L1373-L1397